### ISSUE TITLE
Feature: Badge count

### ISSUE THREAD

**Issue by raix** [raix/push#45]
Add automated badge counter.

There are two different ways of thinking about counters:
* Non-specific messages increments a "count" - this is reset on app open
* Trackable messages sets an id in the "counter" object - The app actively sets id as read

The badge is the sum of these two and is kept updated automatically.

We want an option to set a limit on the sum of unread messages, this could be set as one of three:
* Total unread limit - the limit of the sum of both counters
* Total unread of Non-specific
* Total unread of Trackable

If the limit is reached the server will skip messages for that user, at the same time we want a `force` option on send - but this should only be allowed on the server send api.

**Comment by dennisharrison** [comment]
I like having manual control over the badge number.  I can see where most apps would want this automatically to happen for them though.  

Can you explain more about the limit being reached and server skip messages for that user with a force send option?

**Comment by raix** [comment]
eg. if a user have 40 unread messages - it would in some apps be unlikely that the user actually uses the app - the limit helps protect both client and server from spamming.

We could perhaps have this feature as a separate package

**Comment by dennisharrison** [comment]
Ahh, so a limit of unread messages before auto-ignoring that user?  That sounds like a good addition.  Should this be an additional package or a config flag?

**Comment by raix** [comment]
Not sure - I mean it's a general enough feature to have it as config flag - at the same time it might be easier to maintain as a separate package.

**Comment by dennisharrison** [comment]
It would be integral enough to be included in the main package and set as a config flag.  You're essentially transforming the default behavior of a push notification.  

That said, I'd really like to learn how you would solve some of the componentization issues of having this as a separate package - so I'm interested in tagging along.

**Comment by MoeBadr** [comment]
looking forward to seeing automated badge counter in 2.7.0. I love this package, It saved me hours of work. Thank you so much guys :+1:

**Comment by alpjor** [comment]
++ to this feature

**Comment by guilhermedecampo** [comment]
+1 for this feature. 

Specially because I couldn't find a way to update the badge when the app is in background :disappointed:

**Comment by challett** [comment]
If you have a count available that you want to set the badge to, you can specify it in a badge key inside of your Push.send() call.  This feature is a little hidden in the documentation but it makes it possible to update the badge while the app is in the background

```
Push.send({
     from:'app', 
     title:'updateBadge',
     text:'your badge count is now 5' ,
     badge:5 ,
     query {}
});
```

**Comment by guilhermedecampo** [comment]
:open_mouth: Will test this now @challett ! Thanks for helping.

**Comment by guilhermedecampo** [comment]
Yes, adding badge solve my problem thanks @challett ! :+1:

**Comment by Tim-W** [comment]
Any updates on automatic badge counter?

**Comment by boblikesoup** [comment]
Yes I'd like to know how to add a badge at 0 and increment if one already exists.  The documentation isn't clear to me.

Right now we have something like this that isn't incrementing past 1:
  badge = 0
  if (hasBadge) {badge = 1}
Push.send({...badge: badge})

**Comment by challett** [comment]
@boblikesoup It looks like you're checking if the badge exists and then setting it to one if it does.  The `badge: <number>` flag sets the badge to  `<number>`, it doesn't increment it by `<number>`.  You need a way of querying the number that the badge should be set to.

**Comment by boblikesoup** [comment]
Oooh k thanks.  So there's no way to just increment the number detected on the device?

**Comment by challett** [comment]
I've submitted a pull request to better explain this in the documentation #96

**Comment by raix** [comment]
For now theres only a `Push.setBadge(count);`
https://github.com/raix/push/blob/master/lib/client/cordova.js#L16

It's not yet implemented on the server:
https://github.com/raix/push/blob/master/lib/server/push.api.js#L11

**Comment by mitiaptest** [comment]
On app launch its resting to badge as 0. No badge. how to keep as it is badge on app lauch ?

**Comment by Tzikas** [comment]
Was this issue resolved?  I'm trying to reset the badge, to zero when the app is opened, and I'm incrementing the badge, until then using badge++.  As of now the app just continues to increment regardless of whether or not i open it.

**Comment by Tzikas** [comment]
http://stackoverflow.com/questions/34149089/meteor-how-can-i-detect-if-an-app-has-been-opened
